content/docs/handling-events.md

```diff
@@ -139,3 +139,17 @@ class LoggingButton extends React.Component {
 ```
 
 The problem with this syntax is that a different callback is created each time the `LoggingButton` renders. In most cases, this is fine. However, if this callback is passed as a prop to lower components, those components might do an extra re-rendering. We generally recommend binding in the constructor or using the class fields syntax, to avoid this sort of performance problem.
+
+## Passing arguments to event handlers
+
+Inside a loop it is common to want to pass a param to an event handler. For example if `i` is the row id:
+
+```js
+<button onClick={() => this.deleteRow(i)}>Delete Row</button> 
+```
+
+or alternatively (especially if you want to avoid triggering a re-render in a child component):
```

Comment:
This line isn't actually true ^
```js
function foo() {}
const ctx = {};
const a = foo.bind(ctx);
const b = foo.bind(ctx);
console.log(a == b); // false
```

Binding in the constructor is a way to avoid unnecessary updates but that's not an option when you want to pass in-scope variables like this.

That being said, I think we _should_ still show the `.bind()` form for people who prefer to write vanilla ES5 but it doesn't perform differently in terms of `shouldComponentUpdate` (at least not that I'm aware of).

One way to avoid unnecessary renders would be:
```jsx
class Parent extends Component {
  deleteRow = (index) => {
    // Do stuff...
  };

  render() {
    // Render stuff, including
    <DeleteRowButton index={i} deleteRow={this.deleteRow} />
  }
}

class DeleteRowButton extends Component {
  onClick = () => {
    this.props.deleteRow(this.props.index);
  };

  render() {
    return <button onClick={this.onClick}>Delete Row</button>;
  }
}
```

But all of this ^ is overkill for a simple `<button>`. It would only make sense to do if the component you're rendering is expensive (eg has as lot of nested children or complex logic).

Reply:
i think the examples conflate the method of passing params to events (where this pattern is valid) with passing params within loops (where this pattern is not valid as you point out). I should be clearer there AND also provide the hint to pass by props as newbies can overlook that simple solution. I agree the rerendering thing is overkill and if the reader is worrying about perf to that extent they probably wouldnt be reading this section of the docs anyway. Was only including it because it was suggested in the issue raised.

Reply:
I added some comments about how and when to avoid binds to the FAQ page (#43). It's probably OK to keep it high-level here.

Reply:
> It's probably OK to keep it high-level here.

Agreed.

Reply:
Parsimony is a virtue when it comes to docs. I will just delete the problematic line of code so as not to mislead/confuse.

Reply:
I understand your thinking, but I actually disagree with the decision you guys ended up making. I came to this thread specifically because I was confused why there is a warning about re-renders for `onClick={(e) => this.handleClick(e)}` but not for `onClick={(e) => this.deleteRow(id, e)}`.

Also, being that I'm a relative newbie to React, I didn't fully understand the ramifications of these potential re-renders. The discussion here actually helped me get a sense of how serious a problem they might be from a performance standpoint.

I understand not wanting to pollute the documentation with too much detail, but it would be nice to put the extra details in something akin to footnotes.